src/complex/Utilities/ParallelAlgorithmUtilities.hpp

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include <stdexcept>
+
+#include "complex/Common/Types.hpp"
+
+namespace complex
+{
+template <class ParallelRunnerT>
+struct ParallelRunner
```

Comment:
```suggestion
class ParallelRunner
```
Since this has a `private` section, we use `class` by convention.